File: Scala/single-number-ii.scala
object Solution {
    def singleNumber(nums: Array[Int]): Int = {
        
        var num:Int = 0
        
        for(i <- 0 until 32){
            var vsum = 0
            for(vnum <- nums){
                vsum += ((vnum>>i) & 1)
            }
            vsum = vsum%3
            num |= (vsum<<i)
        }
        
        num
    }
}

File: Python/single-number-ii.py
class Solution:
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        num=0
        
        for i in range(32):
            vsum = 0
            for vnum in nums:
                vsum += ((vnum>>i) & 1)
            
            vsum = vsum%3
            num |= (vsum<<i)
        
        return num
